### PR TITLE
Fix igms2fd to work with the new changes

### LINKED ISSUE #1384: ImportGMS2fontdata script compile error
### Describe the bug

Just like the pic show
image

### Reproducing steps

Scripts/Community Scripts/ImportGMS2fontdata.csx

### Setup Details

1.UndertaleModTool v:0.5.1.0
2.Windows 10

### LINKED ISSUE #1431: About the ImportGMS2FontData.csx script
### Describe the bug

Error CS0117 occurs on lines 185 and 186 in the script, which can be executed after deletion

### Reproducing steps

.

### Setup Details

.

### LINKED ISSUE #1438: CS0117 bug when use ImportGMS2FontData.csx
### Describe the bug

Screenshot (550)

### Reproducing steps

1. Open ImportGMS2FontData.csx
2. See that error

### Setup Details

1.UndertaleModTool v0.5.1.0
2.Window 11
3.Deltarune

### LINKED ISSUE #1497: About the Community Scripts"ImportGMS2FontData.csx"
### Describe the bug

57e41e272e21bcf37e36b3fefe21b27

D:\hanhua\UTMT\Scripts\Community Scripts\ImportGMS2FontData.csx(185,9): error CS0117: “UndertaleFont.Glyph.GlyphKerning”未包含“Other”的定义

### Reproducing steps

image

I clicked this button and then an error window appeared.

### Setup Details

I have changed several versions, but they all have the same error. The following image shows the versions I have downloaded.

image

At the same time, both of these two links are no longer working.

image

### PR DESCRIPTION
**Pull request by Dobby233Liu**
## Description
Fixes #1384; fixes #1431; fixes #1438; fixes #1497

### CODE REVIEW

**Comment by VladiStep** [comment]
Did you accidentally replace all line break characters?

**Comment by Dobby233Liu** [comment]
Sorry haven't come back to this since forever, will try to fix it later

**Comment by Dobby233Liu** [comment]
I'm done I think

File: UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx [hunk 1]
```diff
@@ -52,13 +52,18 @@ Try renaming the correct texture file to
...
-  This was an attempt to get fonts that this script creates appear in a specific 2022.3 game, but was proved unnecessary, as the problem was caused by something else
+  If true, the script will attempt to add the new font (if any) and the new font glyph texture
+  that it created to a texture group
+  This was an attempt to get fonts that this script creates appear in a specific 2022.3 game,
+  but it was proved unnecessary as the problem was caused by something else
 */
-bool attemptToFixFontNotAppearing = false; // Data.GM2022_3;
-// Default to putting the font into the default texgroup
-UndertaleTextureGroupInfo fontTexGroup = Data.TextureGroupInfo.ByName("Default");
+bool attemptToFixFontNotAppearing = tginExists && false; // Data.GM2022_3;
```
Comment (Miepee) [inline UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx:66]: why was the `tginExists` added to here?

File: UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx [hunk 2]
```diff
@@ -52,13 +52,18 @@ Try renaming the correct texture file to
 and putting it in the same directory as the .yy file."
     );
 
+bool tginExists = Data.TextureGroupInfo != null;
```
[the remark this answers is not in the document]
Comment (Dobby233Liu) [reply UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx:55]: Acceptable

File: UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx (hunk 1 above)
Comment (Dobby233Liu) [reply UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx:66]: :thinking: Should I remove

Comment (Miepee) [reply UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx:66]: Ah, was not aware as in the previous iteration it didnt check for it. feel free to keep it then

File: UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Dobby233Liu) [reply UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx:55]:
```suggestion
bool tginExists = Data.TextureGroupInfo is not null;
```

File: UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx (hunk 1 above)
Comment (Dobby233Liu) [reply UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx:66]: I decide to keep it

**Review by Miepee — APPROVED** [review]

**Pull request merged**